actionpack/lib/action_controller/metal/strong_parameters.rb

```diff
@@ -580,6 +580,22 @@ def stringify_keys # :nodoc:
       dup
     end
 
+    def method_missing(method_sym, *args, &block)
+      if @parameters.respond_to?(method_sym)
+        message = <<-DEPRECATE.strip_heredoc
+          Method #{ method_sym } is deprecated and will be removed in Rails 5.1. StrongParameters no longer inherit from
+          hash. Using this deprecated behavior exposes potential security problems. if you continue to use this method
+          you may be creating a security vulunerability in your app that can be exploited. Instead, consider using one
+          of these public methods that will not be deprecated:
```

Comment:
I feel like this suggestion and list of public methods is not needed, but not really strongly against it. Maybe a link to http://edgeapi.rubyonrails.org/classes/ActionController/Parameters.html?